README.md

```diff
@@ -0,0 +1,705 @@
+- QUAY_IO_CLI_PASSWORD
+
+*testing* mode:
+- TEST_QUAY_IO_USERNAME
+- TEST_QUAY_IO_CLI_PASSWORD
+
+#### Step: Login to Ghcr.io
+
+The [docker/login-action@v3](https://github.com/docker/login-action) is used. The following secrets are used:
+
+*production* mode:
+- GIT_HUB_USERNAME
+- GIT_HUB_TOKEN
+
+*testing* mode:
+- TEST_GIT_HUB_USERNAME
+- TEST_GIT_HUB_TOKEN
+
+#### Step: Generate tags and prepare metadata to build and push
+
+The [docker/metadata-action@v5](https://github.com/docker/metadata-action) is used to generate tags, labels and annotations for images. Here is an example of AlmaLinux 8 minimal image's tags:
+```json
+"tags": [
+          "docker.io/***/8-minimal:latest",
+          "docker.io/***/8-minimal:8",
+          "docker.io/***/8-minimal:8.9",
+          "docker.io/***/8-minimal:8.9-20240319",
+          "quay.io/***/8-minimal:latest",
+          "quay.io/***/8-minimal:8",
+          "quay.io/***/8-minimal:8.9",
+          "quay.io/***/8-minimal:8.9-20240319",
+        ],
+```
+#### Step: Build images
+
+The [docker/build-push-action@v5](https://github.com/docker/build-push-action) is used to build images. This step builds the images from corresponded [`Containerfile`](https://github.com/AlmaLinux/container-images/tree/main/Containerfiles), for specified `env.platforms` and using tags from the previous step. After the successful building, the images are loaded into docker, and not yet pushed because they need to be tested. AlmaLinux 8 minimal images buildx looks like:
+```sh
+/usr/bin/docker buildx build --file ./Containerfile.minimal ... \
+ --platform linux/amd64,linux/ppc64le,linux/s390x,linux/arm64 \
+ --provenance false ... \
+ --tag docker.io/***/8-minimal:latest --tag docker.io/***/8-minimal:8 --tag docker.io/***/8-minimal:8.9 --tag docker.io/***/8-minimal:8.9-20240319 --tag quay.io/***/8-minimal:latest --tag quay.io/***/8-minimal:8 --tag quay.io/***/8-minimal:8.9 --tag quay.io/***/8-minimal:8.9-20240319 \
+ --load \
+ --metadata-file /home/runner/work/_temp/docker-actions-toolkit-*/metadata-file \
+ https://github.com/***/container-images.git#270a6d3fd433cfa6c3e1fff5896a92d1ae2896be:Containerfiles/8
+```
+`provenance: false` is to disable the [Provenance attestations](https://docs.docker.com/build/attestations/slsa-provenance/) as Quay.io registry doesn't support such kind of images data.
+
+#### Step: Test images
+
+As every image is loaded into docker, so it can be tested separately for each type and platform. Docker run images by digest:
+```sh
+docker run --platform=${platform} ${{ steps.build-images.outputs.digest }}
+```
+
+#### Step: Push images to Client Library
+
+The [docker/build-push-action@v5](https://github.com/docker/build-push-action) is used. This step pushes built images into *Client Library*. The options are the same as for **Build images** step.
+
+
+#### Step: Change date stamp in Containerfile (default and minimal only)
+
+> The step is skipped if image type is not 'default' or 'minimal'
```

Comment:
❗ Skip this step if the image type is not 'default' or 'minimal'.

Reply:
Done.